### PR TITLE
feat: add mongo db integration

### PR DESCRIPTION
**Pull request by pietro-tota**
#### List of Changes
Add mongo db integration for perform query on the payment wallet db 

#### Motivation and Context
This pr add logic to integrate with Mongo DB (payment wallet CosmosDB with mongo api) in order to perform queries against wallet DB

#### How Has This Been Tested?
Local service run pointing to local mongo db in docker compose performing query to retrieve wallet data and print it to standard output (as a proof of concept for integration)

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/cosmosdb/cosmosdb_client.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+func GetMongoClient() (*mongo.Client, error) {
+	uri, err := utils.ReplaceEnvPropertiesInString(MongoConnectionString)
+	if err != nil {
+		errorMessage := fmt.Sprintf("Cannot instantiate Mongo client, error building connection string: [%s]", err)
+		log.Fatal(errorMessage)
+		return nil, errors.New(errorMessage)
+	}
+	additionalMongoConnectionProperties := utils.GetEnvVariableOrDefault("MONGO_ADDITIONAL_CONNECTION_PROPERTIES", "")
+	mongoConnectionString := *uri + additionalMongoConnectionProperties
+	client, err := mongo.Connect(context.TODO(), options.Client().
```
Comment (giovanniberti) [inline internal/cosmosdb/cosmosdb_client.go:25]: From reading the [documentation](https://pkg.go.dev/context#Background) I think we can safely switch this to a background context, wdyt?

```suggestion
	client, err := mongo.Connect(context.Background(), options.Client().
```

File: internal/cosmosdb/cosmosdb_client.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	utils "pagopa.it/pagopa-payment-wallet-helpdesk-service/internal/utils"
+)
+
+const MongoConnectionString = "mongodb://${MONGO_USERNAME}:${MONGO_PASSWORD}@${MONGO_HOSTNAME}:${MONGO_PORT}/${MONGO_DB_NAME}?ssl=${MONGO_SSL_ENABLED}"
+
+func GetMongoClient() (*mongo.Client, error) {
+	uri, err := utils.ReplaceEnvPropertiesInString(MongoConnectionString)
+	if err != nil {
+		errorMessage := fmt.Sprintf("Cannot instantiate Mongo client, error building connection string: [%s]", err)
+		log.Fatal(errorMessage)
```
Comment (giovanniberti) [inline internal/cosmosdb/cosmosdb_client.go:20]: [This kills the server](https://pkg.go.dev/log#Fatal), maybe it's better to let the caller handle errors?

File: tmp/build-errors.log [hunk 3]
```diff
@@ -0,0 +1 @@
+exit status 2exit status 2exit status 2exit status 2
```
Comment (giovanniberti) [inline tmp/build-errors.log:1]: Stray file? :)

Comment (pietro-tota) [reply tmp/build-errors.log:1]: ops, my bad :)

File: internal/cosmosdb/cosmosdb_client.go (hunk 1 above)
Comment (pietro-tota) [reply internal/cosmosdb/cosmosdb_client.go:25]: I was thinking that the context used here should be the one coming from the http request, so here I set the TODO one as a placeholder for future replacing.
I'm refactoring it in order to accept context as input parameter in order for the http request context to be used here as input, what do you think?

File: internal/cosmosdb/cosmosdb_client.go (hunk 2 above)
Comment (pietro-tota) [reply internal/cosmosdb/cosmosdb_client.go:20]: Yes, you're right, fixing it

File: tmp/build-errors.log (hunk 3 above)
Comment (pietro-tota) [reply tmp/build-errors.log]: fixed with https://github.com/pagopa/pagopa-payment-wallet-helpdesk-service/pull/5/commits/fa1ab87cb24ba10c427aeec4100620059211b299

File: internal/cosmosdb/cosmosdb_client.go (hunk 2 above)
Comment (pietro-tota) [reply internal/cosmosdb/cosmosdb_client.go]: fixed with https://github.com/pagopa/pagopa-payment-wallet-helpdesk-service/pull/5/commits/f9c6d2ef25bb25ee1729f27267002c54000897b3

File: internal/cosmosdb/cosmosdb_client.go (hunk 1 above)
Comment (pietro-tota) [reply internal/cosmosdb/cosmosdb_client.go]: fixed with https://github.com/pagopa/pagopa-payment-wallet-helpdesk-service/pull/5/commits/67eecf51082db71dea99970b5f804a10a2bf7fe9

**Pull request merged**